dist/crawl_links.js

```diff
@@ -0,0 +1,40 @@
+const intersectRect = function (r1, r2) {
```

Comment:
Should we add a comment at the top of this file to explain what we're doing and why? A lot of info in the PR, but nothing in here and am sure we'll soon forget why this is here.